src/requestManager.cpp

```diff
@@ -428,6 +428,15 @@ bool CUnknownObj::AddSource(CNode *from)
     return false;
 }
 
+void CRequestManager::RequestCorruptedBlockByHash(const uint256 &blockHash)
+{
+    // set it to MSG_BLOCK here but it should get overwritten in RequestBlock
+    CInv obj(MSG_BLOCK, blockHash);
+    std::vector<CInv> vGetBlocks;
+    vGetBlocks.push_back(obj);
+    AskForDuringIBD(vGetBlocks, nullptr);
```

Comment:
currently you can not pass a nullptr to askforduringibd()...you could easily add that though.

also a nit, I think you could just shorten the function names and take out the "ByHash"...I think it's somewhat obvious and only makes the name overly long IMO.

Reply:
fixed issue with nullptr, thanks.